socorro/external/es/crashstorage.py

```diff
@@ -138,22 +143,91 @@ def _submit_crash_to_elasticsearch(self, connection, crash_document):
             index_creator.create_socorro_index(es_index)
 
         # Submit the crash for indexing.
-        try:
-            connection.index(
-                index=es_index,
-                doc_type=es_doctype,
-                body=crash_document,
-                id=crash_id
-            )
+        success = False
+        # Don't retry more than 5 times. That is to avoid infinite loops in
+        # case of an unhandled exception.
+        retries_pool = [True] * 5 + [False]
+        while not success and retries_pool.pop(0):
+            try:
+                connection.index(
+                    index=es_index,
+                    doc_type=es_doctype,
+                    body=crash_document,
+                    id=crash_id
+                )
+                success = True
+            except elasticsearch.exceptions.TransportError as e:
+                field_name = None
+
+                if 'MaxBytesLengthExceededException' in e.error:
+                    # This is caused by a string that is way too long for
+                    # Elasticsearch.
+                    matches = re.findall(
+                        FIELD_NAME_STRING_ERROR_REGEX, e.error
```

Comment:
By the way, perhaps because I'm weird but the ALL_CAPS style is usually used for simple constants (e.g. strings, integers). 
If you want to define a global regex consider just making it a private looking thing like
```python
_field_name_string_error_regex = re.compile(...)


 class ESCrashStorage(CrashStorageBase):
    ...
```